src/abstractdataframe/iteration.jl

```diff
@@ -67,54 +70,46 @@ julia> df = DataFrame(x=1:4, y=11:14)
 │ 3   │ 3     │ 13    │
 │ 4   │ 4     │ 14    │
 
-julia> collect(eachcol(df))
+julia> collect(eachcol(df, true))
 2-element Array{Pair{Symbol,AbstractArray{T,1} where T},1}:
  :x => [1, 2, 3, 4]
  :y => [11, 12, 13, 14]
-```
-"""
-eachcol(df::T) where T<: AbstractDataFrame =
-    DataFrameColumns{T, Pair{Symbol, AbstractVector}}(df)
-
-"""
-    columns(df::AbstractDataFrame)
-
-Return a `DataFrameColumns` that iterates an `AbstractDataFrame` column by
-column, yielding column vectors.
-
-**Examples**
-
-```jldoctest
-julia> df = DataFrame(x=1:4, y=11:14)
-4×2 DataFrame
-│ Row │ x     │ y     │
-│     │ Int64 │ Int64 │
-├─────┼───────┼───────┤
-│ 1   │ 1     │ 11    │
-│ 2   │ 2     │ 12    │
-│ 3   │ 3     │ 13    │
-│ 4   │ 4     │ 14    │
 
-julia> collect(columns(df))
+julia> collect(eachcol(df, false))
 2-element Array{AbstractArray{T,1} where T,1}:
  [1, 2, 3, 4]
  [11, 12, 13, 14]
 
-julia> sum.(columns(df))
+julia> sum.(eachcol(df, false))
 2-element Array{Int64,1}:
  10
  50
 
-julia> map(columns(df)) do col
+julia> map(eachcol(df, false)) do col
            maximum(col) - minimum(col)
        end
 2-element Array{Int64,1}:
  3
  3
 ```
 """
-columns(df::T) where T<: AbstractDataFrame =
-    DataFrameColumns{T, AbstractVector}(df)
+eachcol(df::T, names::Bool) where T<: AbstractDataFrame =
```

Comment:
I'd use the long function form, and add `@inline` just in case.

Reply:
OK